x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/file/tool/UsersTool.java

```diff
@@ -439,6 +439,10 @@ static String parseUsername(List<String> args, Settings settings) throws UserExc
 
     private static char[] getPasswordHash(Terminal terminal, Environment env, String cliPasswordValue) throws UserException {
         final Hasher hasher = Hasher.resolve(XPackSettings.PASSWORD_HASHING_ALGORITHM.get(env.settings()));
+        if (XPackSettings.FIPS_MODE_ENABLED.get(env.settings()) && hasher.name().toLowerCase(Locale.ROOT).startsWith("pbkdf2") == false) {
```

Comment:
#55544 makes sure our default is sane for FIPS mode. But user can still explicit set the password hashing algorithm to something that is not FIPS compliant. 
Since `PASSWORD_HASHING_ALGORITHM.get(...)` is used in a few other places, rather than checking it at caller side, would it better to have this validation done in its own [validator](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java#L183)?

Reply:
yes, good point! I though about this too when raising the PR and decided against this based on the following: 

- It felt like we would be doing _too_ much in the validator. My idea about the `Validator` interface for `Setting` is that it is used for simple validation checks. 
- But mostly that this is the only place where we call `get()` on it before the node is started ( where `or.elasticsearch.xpack.Security#validateForFips` would kick in. If we add this in the validator of the setting, then we could as well remove this validation from `validateForFips` and if we do that , we could as well move other validations from that method to the respective settings Validator. While this is not something I am very strongly opiniated about, I like that we have the checks together in `validateForFips` and feel like this change is adding a non crucial validation that helps with the user experience

Reply:
I see your point. Having FIPS validation in one place is preferable. All good. Thanks